x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/OptimizerRules.java

```diff
@@ -85,7 +88,17 @@ protected AttributeSet generates(LogicalPlan logicalPlan) {
                 return new AttributeSet(mvExpand.expanded());
             }
             if (logicalPlan instanceof Enrich enrich) {
-                return new AttributeSet(Expressions.asAttributes(enrich.enrichFields()));
+                List<NamedExpression> enrichFields = enrich.enrichFields();
+                AttributeSet generates = new AttributeSet(Expressions.asAttributes(enrichFields));
+                // In case of aliases we generate both the alias and its target.
+                // E.g. in ENRICH policy ON field WITH alias = enrich_field
+                // we generate both `alias` and `enrich_field`.
```

Comment:
This looks incorrect as enrich_field is hidden behind alias.

Reply:
Yeah, it's weird that we have to do this. I found a [place in the physical optimizer ](https://github.com/elastic/elasticsearch/blob/e568f7038daa8791f6f5db92e514cf9d8408418b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/PhysicalPlanOptimizer.java#L122-L123)that needs to handle Enrich in a similarly weird way.

To avoid cementing this weirdness, I'll remove this from the PR, at the cost of having to disable the dependency checker again. I'll follow this up with another PR, after investigating the weird handling of attributes in Enrich.

Reply:
There we go! https://github.com/elastic/elasticsearch/issues/105807